Google.Api.Generator.Tests/ProtoTests/ChildResource/Testing.ChildResource/ChildResourceClient.g.cs

```diff
@@ -16,226 +16,2486 @@
+            {
+                RefAsProjectName = @ref,
+                RepeatedRefAsDeptNames =
+                {
+                    repeatedRef ?? linq::Enumerable.Empty<DeptName>(),
+                },
+            }, callSettings);
+
+        /// <summary>
+        /// </summary>
+        /// <param name="ref">
+        /// </param>
+        /// <param name="repeatedRef">
+        /// </param>
+        /// <param name="callSettings">If not null, applies overrides to this RPC call.</param>
+        /// <returns>A Task containing the RPC response.</returns>
+        public virtual stt::Task<Response> TripleParentMethodAsync(ProjectName @ref, scg::IEnumerable<DeptName> repeatedRef, gaxgrpc::CallSettings callSettings = null) =>
+            TripleParentMethodAsync(new TripleParent
+            {
+                RefAsProjectName = @ref,
+                RepeatedRefAsDeptNames =
+                {
+                    repeatedRef ?? linq::Enumerable.Empty<DeptName>(),
+                },
+            }, callSettings);
+
+        /// <summary>
+        /// </summary>
+        /// <param name="ref">
+        /// </param>
+        /// <param name="repeatedRef">
+        /// </param>
+        /// <param name="cancellationToken">A <see cref="st::CancellationToken"/> to use for this RPC.</param>
+        /// <returns>A Task containing the RPC response.</returns>
+        public virtual stt::Task<Response> TripleParentMethodAsync(ProjectName @ref, scg::IEnumerable<DeptName> repeatedRef, st::CancellationToken cancellationToken) =>
+            TripleParentMethodAsync(@ref, repeatedRef, gaxgrpc::CallSettings.FromCancellationToken(cancellationToken));
+
+        /// <summary>
+        /// </summary>
+        /// <param name="ref">
+        /// </param>
+        /// <param name="repeatedRef">
+        /// </param>
+        /// <param name="callSettings">If not null, applies overrides to this RPC call.</param>
+        /// <returns>The RPC response.</returns>
+        public virtual Response TripleParentMethod(ProjectName @ref, scg::IEnumerable<gax::IResourceName> repeatedRef, gaxgrpc::CallSettings callSettings = null) =>
+            TripleParentMethod(new TripleParent
+            {
+                RefAsProjectName = @ref,
+                RepeatedRefAsResourceNames =
+                {
+                    repeatedRef ?? linq::Enumerable.Empty<gax::IResourceName>(),
+                },
+            }, callSettings);
 
+        /// <summary>
+        /// </summary>
+        /// <param name="ref">
+        /// </param>
+        /// <param name="repeatedRef">
+        /// </param>
+        /// <param name="callSettings">If not null, applies overrides to this RPC call.</param>
+        /// <returns>A Task containing the RPC response.</returns>
+        public virtual stt::Task<Response> TripleParentMethodAsync(ProjectName @ref, scg::IEnumerable<gax::IResourceName> repeatedRef, gaxgrpc::CallSettings callSettings = null) =>
```

Comment:
I can't remember why we have IResourceName overloads here - a TripleName parent is always one of ProjectName, DeptName, OrgName... do we always add an IResourceName overload whenever child_type is specified? Or only when there are multiple parents? (I'd expect this for TripleWildcard, but not TripleParent. I suspect there's a good reason for it - I don't have the doc open right now.)

Reply:
Yes, you're right, these shouldn't be here. The `IResourceName` properties are required in the message partial class, but not as a method overload here. I'll fix this.

Reply:
Just thinking that the partial class `IResourceName` properties should probably be `get`-only if there isn't a wildcard specified. When setting the user should always have a strong-typed resource-name; it's only when getting that the type can't be known.

Reply:
Just to clarify this conversation, Chris and I then chatted offline - one common scenario where you might *not* know the type is copying from one object to another, which is why Chris re-added the setter.